src/java/org/apache/cassandra/index/sai/disk/v1/kdtree/BKDReader.java

```diff
@@ -158,6 +158,16 @@ public IteratorState(DocMapper docMapper, boolean isAscending, BKDReader.Interse
                 // Inclusive on both ends because min/maxLeafFP return the offsets to the leaves
                 // that intersect the query (i.e. contain at least one point matching the query) and
                 // possible range query exclusiveness is handled by the query object.
+
+                // if descending, we need to swap the leaf bounds because the order of the map is reversed. We used our
+                // query to find minLeaf/maxLeaf, which retains the natural ascending order.
```

Comment:
Great catch. It's interesting to note that the descending map flips the from and to as well.

```java
    static final class DescendingSubMap<K,V>  extends NavigableSubMap<K,V> {

        public NavigableMap<K,V> subMap(K fromKey, boolean fromInclusive,
                                        K toKey,   boolean toInclusive) {
            if (!inRange(fromKey, fromInclusive))
                throw new IllegalArgumentException("fromKey out of range");
            if (!inRange(toKey, toInclusive))
                throw new IllegalArgumentException("toKey out of range");
            return new DescendingSubMap<>(m,
                                          false, toKey,   toInclusive,
                                          false, fromKey, fromInclusive);
        }
    }
```

Not that we need to do it, but an alternate solution would be to defer the call to make the map a descending map.